### DISCUSSION TITLE
Error response from daemon: could not select device driver "nvidia" with capabilities: [[gpu]]

### DISCUSSION THREAD

**Discussion by grug0r** [blakeblackshear/frigate#5536] [Ask A Question]
Server Specs:
Windows 2019 Server Hyper-V
Ubuntu 18.04.6 LTS Virtual Machine
Nvidia Quadro K2200 

lspci
  857d:00:00.0 VGA compatible controller: NVIDIA Corporation GM107GL [Quadro K2200] (rev a2)

lshw -C Display
  description: VGA compatible controller
  product: GM107GL [Quadro K2200]
  vendor: NVIDIA Corporation
  physical id: 1
  bus info: pci@857d:00:00.0
  version: a2
  width: 64 bits
  clock: 33MHz
  capabilities: pm msi pciexpress vga_controller bus_master cap_list
  configuration: driver=nvidia latency=0
  resources: iomemory:f0-ef iomemory:f0-ef irq:0 memory:c1000000-c1ffffff memory:fe0000000-fefffffff memory:ff0000000-ff1ffffff

+-----------------------------------------------------------------------------+
| NVIDIA-SMI 525.60.13    Driver Version: 525.60.13    CUDA Version: 12.0     |
|-------------------------------+----------------------+----------------------+
| GPU  Name        Persistence-M| Bus-Id        Disp.A | Volatile Uncorr. ECC |
| Fan  Temp  Perf  Pwr:Usage/Cap|         Memory-Usage | GPU-Util  Compute M. |
|                               |                      |               MIG M. |
|===============================+======================+======================|
|   0  Quadro K2200        Off  | 0000857D:00:00.0 Off |                  N/A |
| 42%   46C    P0     2W /  39W |      0MiB /  4096MiB |      1%      Default |
|                               |                      |                  N/A |
+-------------------------------+----------------------+----------------------+

+-----------------------------------------------------------------------------+
| Processes:                                                                  |
|  GPU   GI   CI        PID   Type   Process name                  GPU Memory |
|        ID   ID                                                   Usage      |
|=============================================================================|
|  No running processes found                                                 |
+-----------------------------------------------------------------------------+

All appears good above?

My docker-compose.yml file:
[docker-compose.yml](https://pastebin.com/8zztr6nF)

Running the command:
sudo docker-compose up
I get the following:
Error response from daemon: could not select device driver "nvidia" with capabilities: [[gpu]]

Grateful for any ideas what I'm doing wrong?

**Comment by NickM-27** [comment]
You'll most likely need to add the env var `NVIDIA_VISIBLE_DEVICES` with a value of `all`

In 0.12 this isn't necessary anymore

**Comment by grug0r** [comment]
I tried your suggestion, received same result.  I pulled down 0.12.0-beta8 and also same result.

**Reply by NickM-27** [reply]
Maybe try with count commented out and uncomment ids

**Reply by grug0r** [reply]
Thanks for the suggestion.. this helped.

**Comment by NateMeyer** [comment]
Do you have the nvidia-container-runtime installed? See the  [docker docs](https://docs.docker.com/config/containers/resource_constraints/#gpu)

**Reply by grug0r** [reply]
I did not. Following installation I'm now seeing progress passed the original error. Though now have a different issue running [sudo docker-compose up](https://pastebin.com/JYGduRAe)

Here's my [frigate.yml](https://pastebin.com/JztMvxVW) and [docker-compose.yml](https://pastebin.com/Ex4ekXyS) files for review. I tried to keep them to the bare essentials.

**Reply by NickM-27** [reply]
I saw another user had this issue because they weren't defining all capabilities. 

`NVIDIA_DRIVER_CAPABILITIES="compute,video,utility"`

The container in 0.12 has it set internally but you may want to try adding it to your docker compose just to see.

**Reply by NateMeyer** [reply]
See #5166.  Can you verify libnvcuvid is installed on your host system?

**Reply by grug0r** [reply]
Hi Nate,

Output of apt install nvidia-container-runtime
nvidia-container-runtime is already the newest version (3.12.0-1).

Output of [ldconfig -p](https://pastebin.com/n9aa7yim)
I note there isn't a libnvcuvid in the list.

Output of [dpkg -l '*nvidia*'](https://pastebin.com/3fsFAkcB)

Am I missing a Dependency?

PS. I have not tested Nick's suggestion as yet.
